In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('title.basics.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB


what is the title basic tsv is about?
has 5631729 entries i.e 5.6 million entries of ?
relationship with genres is 1 : many


In [108]:
titles = data

In [109]:
titles.head()

tconst titleType            primaryTitle           originalTitle  \
0  tt0000001     short              Carmencita              Carmencita   
1  tt0000002     short  Le clown et ses chiens  Le clown et ses chiens   
2  tt0000003     short          Pauvre Pierrot          Pauvre Pierrot   
3  tt0000004     short             Un bon bock             Un bon bock   
4  tt0000005     short        Blacksmith Scene        Blacksmith Scene   

   isAdult startYear endYear runtimeMinutes                    genres  
0        0      1894      \N              1         Documentary,Short  
1        0      1892      \N              5           Animation,Short  
2        0      1892      \N              4  Animation,Comedy,Romance  
3        0      1892      \N             \N           Animation,Short  
4        0      1893      \N              1              Comedy,Short

In [19]:
titles['originalTitle'].nunique()

2931498

In [14]:
titles['titleType'].unique()

array(['short', 'movie', 'tvMovie', 'tvSeries', 'tvEpisode', 'tvShort',
       'tvMiniSeries', 'tvSpecial', 'video', 'videoGame'], dtype=object)

In [9]:
titles['tconst'].nunique()

5631729

In [17]:
titles['genres'].unique()
titles['genres'].head()

0           Documentary,Short
1             Animation,Short
2    Animation,Comedy,Romance
3             Animation,Short
4                Comedy,Short
Name: genres, dtype: object

In [110]:
titles['titleType'].value_counts()

tvEpisode       3874830
short            665570
movie            510690
video            222234
tvSeries         159635
tvMovie          125774
tvMiniSeries      24791
videoGame         22915
tvSpecial         16250
tvShort            9040
Name: titleType, dtype: int64

In [114]:
data1 = pd.read_csv('name.basics.tsv',sep='\t')

has 32 million entries

In [115]:
peopleWhoWorkInMovies = data1

In [119]:
peopleWhoWorkInMovies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9206045 entries, 0 to 9206044
Data columns (total 6 columns):
nconst               object
primaryName          object
birthYear            object
deathYear            object
primaryProfession    object
knownForTitles       object
dtypes: object(6)
memory usage: 421.4+ MB


In [121]:
peopleWhoWorkInMovies.birthYear.value_counts()

\N      8730452
1980       8226
1979       8130
1978       8116
1981       8008
1982       7804
1977       7587
1976       7580
1983       7312
1975       7291
1974       7178
1984       7158
1970       7096
1971       6994
1972       6933
1973       6801
1985       6743
1986       6636
1969       6515
1968       6099
1967       5798
1965       5771
1987       5767
1964       5743
1966       5643
1963       5467
1988       5410
1962       5401
1961       5146
1947       5053
         ...   
0163          1
1413          1
1644          1
0770          1
1291          1
1733          1
1625          1
1597          1
1745          1
1677          1
1648          1
1631          1
1569          1
1765          1
1265          1
1468          1
1515          1
1562          1
1414          1
1532          1
1314          1
1586          1
1754          1
1207          1
1600          1
1752          1
0037          1
1415          1
0021          1
1681          1
Name: birthYear, Length:

In [118]:
data1 = pd.read_csv('title.principals.tsv',sep='\t')

In [122]:
principal = data1

In [123]:
principal.head()

tconst  ordering     nconst         category                      job  \
0  tt0000001         1  nm1588970             self                       \N   
1  tt0000001         2  nm0005690         director                       \N   
2  tt0000001         3  nm0374658  cinematographer  director of photography   
3  tt0000002         1  nm0721526         director                       \N   
4  tt0000002         2  nm1335271         composer                       \N   

    characters  
0  ["Herself"]  
1           \N  
2           \N  
3           \N  
4           \N

In [129]:
principal.tconst.value_counts(ascending=False).unique()

array([10,  9,  8,  7,  6,  5,  4,  3,  2,  1])

In [136]:
principal.job.value_counts(ascending=False)

\N                                                     26918368
producer                                                1876783
writer                                                   693415
creator                                                  496450
written by                                               336796
story                                                    159174
director of photography                                  143642
screenplay                                               136989
head writer                                               95814
adaptation                                                83734
co-director                                               71186
novel                                                     42195
teleplay                                                  41150
series director                                           37010
dialogue                                                  36625
script                                  

In [130]:
principal.categorytegory.value_counts(ascending=False)

actor                  7668975
actress                5560782
self                   5495951
writer                 4100995
director               3611035
producer               1918242
composer               1187555
cinematographer        1154846
editor                 1078652
production_designer     261679
archive_footage         187035
archive_sound             1949
Name: category, dtype: int64

In [26]:
principal['category'].unique()

array(['self', 'director', 'cinematographer', 'composer', 'producer',
       'editor', 'actor', 'actress', 'writer', 'production_designer',
       'archive_footage', 'archive_sound'], dtype=object)

In [29]:
principal['characters'].unique()

array(['["Herself"]', '\\N', '["Blacksmith"]', ..., '["Andy Westfall"]',
       '["Yirga"]', '["Mike van Loo"]'], dtype=object)

shows for a movie the nconst has played what character in the movie

In [30]:
principal['tconst'].nunique()

5047848

In [37]:
data1 = pd.read_csv('title.crew.tsv',sep='\t')

In [38]:
crew = data1

In [39]:
crew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5723534 entries, 0 to 5723533
Data columns (total 3 columns):
tconst       object
directors    object
writers      object
dtypes: object(3)
memory usage: 131.0+ MB


In [100]:
crew.head()

tconst  directors writers
0  tt0000001  nm0005690      \N
1  tt0000002  nm0721526      \N
2  tt0000003  nm0721526      \N
3  tt0000004  nm0721526      \N
4  tt0000005  nm0005690      \N

In [41]:
data1 = pd.read_csv('title.ratings.tsv',sep='\t')

In [42]:
ratings = data1

In [43]:
ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 917181 entries, 0 to 917180
Data columns (total 3 columns):
tconst           917181 non-null object
averageRating    917181 non-null float64
numVotes         917181 non-null int64
dtypes: float64(1), int64(1), object(1)
memory usage: 21.0+ MB


In [44]:
ratings.head()

tconst  averageRating  numVotes
0  tt0000001            5.8      1478
1  tt0000002            6.4       176
2  tt0000003            6.6      1108
3  tt0000004            6.4       107
4  tt0000005            6.2      1813

In [45]:
ratings.describe()

averageRating      numVotes
count  917181.000000  9.171810e+05
mean        6.918489  9.705193e+02
std         1.401053  1.553299e+04
min         1.000000  5.000000e+00
25%         6.200000  9.000000e+00
50%         7.100000  2.000000e+01
75%         7.900000  7.700000e+01
max        10.000000  2.064369e+06

In [93]:

data1 = pd.read_csv('title.akas.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [94]:
alsoKnownAs = data1

In [132]:
alsoKnownAs.titleId.value_counts(ascending=False)

tt8185052    129
tt0168366    115
tt8052676     88
tt0076759     79
tt8232202     78
tt8187756     77
tt0060196     73
tt0057012     68
tt0080684     67
tt0029583     64
tt0088247     63
tt0086190     63
tt0071562     62
tt2488496     61
tt0058461     61
tt0064116     60
tt0068646     59
tt0080339     59
tt0082971     59
tt0044706     58
tt0111161     58
tt0052357     57
tt0245429     57
tt0047478     57
tt0096874     57
tt0367882     57
tt0028950     57
tt0064208     56
tt0047396     56
tt0073195     56
            ... 
tt1604408      1
tt6025896      1
tt0745627      1
tt4866710      1
tt6760312      1
tt3788922      1
tt0044678      1
tt0907126      1
tt3632686      1
tt0590394      1
tt5424634      1
tt2110365      1
tt3825272      1
tt7283236      1
tt0286242      1
tt7183820      1
tt6750864      1
tt0538874      1
tt0995145      1
tt1016822      1
tt5901648      1
tt2077325      1
tt5054554      1
tt7187408      1
tt7148720      1
tt1864082      1
tt0977207      1
tt2083219     

In [47]:
alsoKnownAs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3484661 entries, 0 to 3484660
Data columns (total 8 columns):
titleId            object
ordering           int64
title              object
region             object
language           object
types              object
attributes         object
isOriginalTitle    object
dtypes: int64(1), object(7)
memory usage: 212.7+ MB


In [48]:
alsoKnownAs.head()

titleId  ordering                      title region language  \
0  tt0000001         1  Carmencita - spanyol tánc     HU       \N   
1  tt0000001         2                 Καρμενσίτα     GR       \N   
2  tt0000001         3                 Карменсита     RU       \N   
3  tt0000001         4                 Carmencita     US       \N   
4  tt0000001         5                 Carmencita     \N       \N   

         types attributes isOriginalTitle  
0  imdbDisplay         \N               0  
1           \N         \N               0  
2           \N         \N               0  
3           \N         \N               0  
4     original         \N               1

In [49]:
alsoKnownAs['titleId'].nunique()

2023643

In [133]:
alsoKnownAs['region'].value_counts()

US      906267
\N      404744
GB      253700
FR      139897
JP      115591
ES      105772
DE      103437
XWW     101703
CA       87597
AR       81500
IT       78541
AU       73243
BR       73243
GR       63540
FI       56416
PT       51747
HU       50524
RU       45531
IN       45248
PL       43039
SE       42271
XWG      41967
MX       41726
DK       40529
NL       27873
TR       27175
SUHH     23834
BE       22870
AT       19456
BG       18861
         ...  
AQ           5
TV           5
CK           4
AS           4
SB           4
XPI          4
GF           4
EH           3
VA           3
WF           3
GD           3
SC           3
GQ           3
XNA          2
ZRCD         2
JE           2
WS           2
SH           2
MS           2
ST           2
KN           2
KI           2
NU           1
XAU          1
AI           1
PW           1
VG           1
MP           1
DJ           1
NR           1
Name: region, Length: 245, dtype: int64

In [134]:
alsoKnownAs['types'].value_counts()

\N                  2550556
imdbDisplay          427658
original             356622
alternative           52300
working               39566
dvd                   18617
tv                    14578
festival              12593
video                 12149
tvvideo                  6
videoworking             5
tvworking                4
dvdimdbDisplay           3
festivalworking          2
dvdworking               1
dvdvideo                 1
Name: types, dtype: int64

In [135]:
alsoKnownAs['language'].value_counts()

\N     3230043
en      138519
ru       21872
fr       18854
bg       17933
tr       15479
sv        6632
he        5430
qbn       4649
sr        3430
cmn       2952
yue       2653
ca        1838
fa        1786
hi        1424
hr        1335
es        1118
tl         885
cs         795
de         785
sl         723
ar         537
nl         536
qbp        496
ta         496
ml         475
te         329
sk         228
bn         224
ur         223
        ...   
da          10
no           7
gd           6
ps           5
uz           5
xh           5
mi           4
vi           4
sd           3
st           3
kk           3
be           2
is           2
ko           2
wo           2
zh           2
tg           2
tn           2
ku           1
goh          1
rm           1
prs          1
rn           1
sq           1
fro          1
myv          1
nqo          1
iu           1
cr           1
ky           1
Name: language, Length: 95, dtype: int64

In [57]:
data1 = pd.read_csv('title.episode.tsv',sep='\t')

In [58]:
episodes = data1

In [59]:
episodes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3874634 entries, 0 to 3874633
Data columns (total 4 columns):
tconst           object
parentTconst     object
seasonNumber     object
episodeNumber    object
dtypes: object(4)
memory usage: 118.2+ MB


In [60]:
episodes.head()

tconst parentTconst seasonNumber episodeNumber
0  tt0041951    tt0041038            1             9
1  tt0042816    tt0989125            1            17
2  tt0042889    tt0989125           \N            \N
3  tt0043426    tt0040051            3            42
4  tt0043631    tt0989125            2            16

In [71]:
episodes.groupby(['parentTconst'])['tconst'].count()

parentTconst
tt0038276       1
tt0039122       5
tt0039123     587
tt0039125      12
tt0040021      65
tt0040027       2
tt0040028      14
tt0040030       1
tt0040032       1
tt0040033       4
tt0040034       4
tt0040036      82
tt0040038       1
tt0040039      10
tt0040040       1
tt0040041     261
tt0040042      78
tt0040045       4
tt0040048    1003
tt0040049     266
tt0040051     467
tt0040052       2
tt0040053    1070
tt0040055       1
tt0040057      21
tt0040058       1
tt0040993      18
tt0040994      23
tt0040995       1
tt0040997       1
             ... 
tt9741848       1
tt9742390      53
tt9742810      26
tt9742936       1
tt9743578       2
tt9743630       4
tt9744774       1
tt9745284       5
tt9747398      21
tt9747414       2
tt9748016       4
tt9748396       5
tt9748472       2
tt9748820       3
tt9749192      19
tt9749378      13
tt9749744       3
tt9749918       1
tt9749982       3
tt9750148       5
tt9751532       1
tt9753030      12
tt9753256       1
tt9755000      

In [78]:
episodes[episodes.parentTconst == 'tt9758034']

tconst parentTconst seasonNumber episodeNumber
3874066  tt9758294    tt9758034            1             1
3874072  tt9758306    tt9758034            1             2
3874081  tt9758326    tt9758034            1             3
3874086  tt9758336    tt9758034            1             4
3874097  tt9758360    tt9758034            1             5
3874102  tt9758370    tt9758034            1             6
3874108  tt9758384    tt9758034            1             7
3874111  tt9758392    tt9758034            1             8
3874116  tt9758402    tt9758034            1             9
3874120  tt9758412    tt9758034            1            10
3874122  tt9758416    tt9758034            1            11
3874123  tt9758418    tt9758034            1            12
3874125  tt9758422    tt9758034            1            13
3874128  tt9758428    tt9758034            1            14
3874129  tt9758430    tt9758034            1            15
3874131  tt9758434    tt9758034            1            16
3874133  tt9758438    tt9758034            1            17
3874135  tt9758442    tt9758034            1            18

In [88]:
ratings[ratings.tconst.isin(episodes.parentTconst.unique())].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 60983 entries, 19682 to 916396
Data columns (total 3 columns):
tconst           60983 non-null object
averageRating    60983 non-null float64
numVotes         60983 non-null int64
dtypes: float64(1), int64(1), object(1)
memory usage: 1.9+ MB


In [92]:
episodes.parentTconst.value_counts(ascending=False)

tt0058796    10541
tt0053494     9667
tt1985601     9502
tt0088580     8911
tt0068120     8697
tt0283794     8651
tt0283767     8104
tt0159881     7859
tt0092325     7636
tt0088512     7633
tt0094481     7081
tt0072584     6795
tt0055708     6746
tt0103434     6695
tt0068069     6420
tt0069658     6353
tt0108977     6043
tt0096597     5903
tt0056758     5870
tt0138228     5584
tt0371487     5186
tt0108979     4664
tt0103569     4656
tt1152300     4601
tt6769314     4601
tt0057741     4443
tt0106053     4238
tt0170980     4069
tt0044298     4052
tt2070564     3978
             ...  
tt2731560        1
tt6113146        1
tt5766950        1
tt7654048        1
tt3668324        1
tt6539592        1
tt4636670        1
tt6233072        1
tt7166874        1
tt6815958        1
tt9071112        1
tt5872906        1
tt2492910        1
tt7707776        1
tt3771468        1
tt6257772        1
tt2300026        1
tt5092982        1
tt4519414        1
tt0268817        1
tt4549714        1
tt0279574   

In [2]:
import pandas as pd
person = pd.read_csv('name.basics.tsv',sep='\t')

In [3]:
person.head()

nconst      primaryName birthYear deathYear  \
0  nm0000001     Fred Astaire      1899      1987   
1  nm0000002    Lauren Bacall      1924      2014   
2  nm0000003  Brigitte Bardot      1934        \N   
3  nm0000004     John Belushi      1949      1982   
4  nm0000005   Ingmar Bergman      1918      2007   

                primaryProfession                           knownForTitles  
0  soundtrack,actor,miscellaneous  tt0072308,tt0053137,tt0043044,tt0050419  
1              actress,soundtrack  tt0038355,tt0071877,tt0037382,tt0117057  
2     actress,soundtrack,producer  tt0059956,tt0057345,tt0054452,tt0049189  
3         actor,writer,soundtrack  tt0078723,tt0072562,tt0080455,tt0077975  
4           writer,director,actor  tt0069467,tt0050976,tt0050986,tt0083922

In [28]:
personKnownForTitles.head(0)

nconst                           knownForTitles
0   nm0000001  tt0072308,tt0053137,tt0043044,tt0050419
1   nm0000002  tt0038355,tt0071877,tt0037382,tt0117057
2   nm0000003  tt0059956,tt0057345,tt0054452,tt0049189
3   nm0000004  tt0078723,tt0072562,tt0080455,tt0077975
4   nm0000005  tt0069467,tt0050976,tt0050986,tt0083922
5   nm0000006  tt0036855,tt0038787,tt0038109,tt0071877
6   nm0000007  tt0037382,tt0034583,tt0043265,tt0033870
7   nm0000008  tt0047296,tt0078788,tt0068646,tt0070849
8   nm0000009  tt0061184,tt0059749,tt0087803,tt0057877
9   nm0000010  tt0029870,tt0055256,tt0035575,tt0031867
10  nm0000011  tt0034167,tt0044706,tt0035211,tt0027996
11  nm0000012  tt0031210,tt0035140,tt0056687,tt0042192
12  nm0000013  tt0060463,tt0055100,tt0053172,tt0048317
13  nm0000014  tt0041452,tt0031381,tt0029843,tt0040806
14  nm0000015  tt0048028,tt0045458,tt0048545,tt0049261
15  nm0000016  tt0069946,tt0091763,tt0080610,tt0096320
16  nm0000017  tt0052311,tt0040367,tt0055031,tt0051201
17  nm0000018  tt0049456,tt0052365,tt0054331,tt0080736
18  nm0000019  tt0056801,tt0047528,tt0053779,tt0071129
19  nm0000020  tt0032551,tt0050083,tt0064116,tt0082846
20  nm0000021  tt0039504,tt0036969,tt0032976,tt0034248
21  nm0000022  tt0025316,tt0031381,tt0026752,tt0052278
22  nm0000023  tt0037059,tt0047522,tt0055031,tt0032138
23  nm0000024  tt0117631,tt0045943,tt0071877,tt0082031
24  nm0000025  tt0117731,tt0100802,tt0119488,tt0112715
25  nm0000026  tt0053125,tt0056923,tt0036613,tt0048728
26  nm0000027  tt0051739,tt0041546,tt0050212,tt0076759
27  nm0000028  tt0038559,tt0040552,tt0040525,tt0036723
28  nm0000029  tt0077800,tt0074802,tt0110138,tt0102122
29  nm0000030  tt0058385,tt0056923,tt0054698,tt0046250
30  nm0000031  tt0063227,tt0082846,tt0032904,tt0043265
31  nm0000032  tt0063442,tt0070723,tt0049833,tt0052618
32  nm0000033  tt0030341,tt0054215,tt0052357,tt0053125
33  nm0000034  tt0043014,tt0046359,tt0074958,tt0065214
34  nm0000035  tt0120647,tt0499549,tt0120746,tt0120338
35  nm0000036  tt0016332,tt0015324,tt0015863,tt0017925
36  nm0000037  tt0045152,tt0043278,tt0041944,tt0041716
37  nm0000038  tt0046912,tt0048728,tt0044706,tt0047396
38  nm0000039  tt0049408,tt0050105,tt0045793,tt0039192
39  nm0000040  tt0072684,tt0120663,tt0062622,tt0066921
40  nm0000041  tt0080979,tt0047478,tt0089881,tt0051808
41  nm0000042  tt0038369,tt0035432,tt0046303,tt0043292
42  nm0000043  tt0034798,tt0034881,tt0038369,tt0034240
43  nm0000044  tt0050468,tt0045793,tt0059825,tt0053793
44  nm0000045  tt0068767,tt0068935,tt0070034,tt0077594
45  nm0000046  tt0031381,tt0055382,tt0044081,tt0033238
46  nm0000047  tt0076085,tt0054749,tt0060121,tt0058335
47  nm0000048  tt0033870,tt0036613,tt0034583,tt0022100
48  nm0000049  tt0057413,tt0383216,tt0085101,tt0054698
49  nm0000050  tt0028772,tt0023969,tt0026778,tt0022158

In [77]:
personKnownForTitles = person[['nconst','knownForTitles']]
frame = pd.DataFrame(personKnownForTitles.knownForTitles.str.split(',').tolist(), index=personKnownForTitles.nconst).stack()
frame = frame.reset_index()[['nconst',0]]
frame.columns = ['nconst','knownForTitles']

In [78]:
frame.head(10)

nconst knownForTitles
0  nm0000001      tt0072308
1  nm0000001      tt0053137
2  nm0000001      tt0043044
3  nm0000001      tt0050419
4  nm0000002      tt0038355
5  nm0000002      tt0071877
6  nm0000002      tt0037382
7  nm0000002      tt0117057
8  nm0000003      tt0059956
9  nm0000003      tt0057345

In [79]:
frame.to_csv('knownForTitles.tsv',sep='\t')

In [41]:
personKnownForTitles = person[['nconst','knownForTitles']]

In [ ]:
frame = pd.concat([pd.Series(row['nconst'], row['knownForTitles'].split(','))              
                    for _, row in personKnownForTitles.iterrows()])

In [42]:
frame = pd.concat([pd.Series(row['nconst'], row['knownForTitles'].split(','))              
                    for _, row in personKnownForTitles.iterrows()])

KeyboardInterrupt: 

In [36]:
frame.head()

tt0072308    nm0000001
tt0053137    nm0000001
tt0043044    nm0000001
tt0050419    nm0000001
tt0038355    nm0000002
dtype: object

In [17]:
pd.concat([pd.Series(row['var2'], row['var1'].split(','))              
                    for _, row in a.iterrows()]).reset_index()

index  0
0     a  1
1     b  1
2     c  1
3     d  2
4     e  2
5     f  2

In [121]:
temp = data[['tconst','genres']]
temp.dropna(inplace=True)
frame = pd.DataFrame(temp.genres.str.split(',').tolist(), index=temp.tconst).stack()
frame = frame.reset_index()[['tconst',0]]
frame.columns = ['tconst','genres']
frame.to_csv('title_genres.tsv', sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [119]:
temp[temp['genres'].apply(lambda x: type(x)!=str)]

Empty DataFrame
Columns: [tconst, genres]
Index: []

In [118]:
temp.tconst.nunique()

5631724

In [ ]:
temp.tconst.nunique()

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


AttributeError: 'str' object has no attribute 'str'

In [84]:
temp.head()

tconst                    genres
0  tt0000001         Documentary,Short
1  tt0000002           Animation,Short
2  tt0000003  Animation,Comedy,Romance
3  tt0000004           Animation,Short
4  tt0000005              Comedy,Short

In [86]:
temp.applymap(lambda x: isinstance(x, (str, float)))

tconst  genres
0          True    True
1          True    True
2          True    True
3          True    True
4          True    True
5          True    True
6          True    True
7          True    True
8          True    True
9          True    True
10         True    True
11         True    True
12         True    True
13         True    True
14         True    True
15         True    True
16         True    True
17         True    True
18         True    True
19         True    True
20         True    True
21         True    True
22         True    True
23         True    True
24         True    True
25         True    True
26         True    True
27         True    True
28         True    True
29         True    True
...         ...     ...
5631699    True    True
5631700    True    True
5631701    True    True
5631702    True    True
5631703    True    True
5631704    True    True
5631705    True    True
5631706    True    True
5631707    True    True
5631708    True    True
5631709    True    True
5631710    True    True
5631711    True    True
5631712    True    True
5631713    True    True
5631714    True    True
5631715    True    True
5631716    True    True
5631717    True    True
5631718    True    True
5631719    True    True
5631720    True    True
5631721    True    True
5631722    True    True
5631723    True    True
5631724    True    True
5631725    True    True
5631726    True    True
5631727    True    True
5631728    True    True

[5631729 rows x 2 columns]

In [80]:
data = pd.read_csv('title.basics.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [81]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB
